crates/factor-outbound-http/src/wasi.rs

```diff
@@ -114,13 +164,198 @@ impl WasiHttpView for WasiHttpImplInner<'_> {
                     self.state.request_interceptor.clone(),
                     self.state.self_request_origin.clone(),
                     self.state.blocked_networks.clone(),
+                    http_clients,
                 )
                 .in_current_span(),
             ),
         ))
     }
 }
 
+#[derive(Clone)]
+struct ConnectOptions {
+    blocked_networks: BlockedNetworks,
+    connect_timeout: Duration,
+}
+
+// We must use task-local variables for these config options when using
+// `hyper_util::client::legacy::Client::request` because there's no way to plumb
+// them through as parameters.  Moreover, if there's already a pooled connection
+// ready, we'll reuse that and ignore these options anyway.
```

Comment:
I'd probably take the tiny theoretical performance hit and use a `OnceLock` for building the clients to avoid this.